### PR TITLE
Add post request

### PR DESCRIPTION
**Pull request by ASands17**
What does this PR do?
- Adds functionality for user to post to local API.
- Adds some methods to activity class.
- Adds tests for activity class.
- Adds tabs to div elements.

Where should your reviewer start?
- Index.html: Line 18
- Activity.js: Line 1
- Activity-test.js: Line 21
- styles.css: Line 127
- scripts.js: Line 5

What (if any) features are you implementing?
- User can now post hydration, activity, and sleep data to local API.
- User can see minutes active and daily miles walked displayed on the screen.

What (if anything) did you refactor?
- Removed unnecessary code from post fetch request.

Were there any issues that arose?
- N/A

Is there anything that you need from your teammate?
- N/A

Any other comments, questions, or concerns?
- N/A

### CODE REVIEW

File: src/Activity.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+  getStrideLengthById(id) {
+    let strideById = this.userData.filter((user) => {
+      return user.id === id;
+    });
+    return strideById[0].strideLength;
+  }
+
+  getDailyMilesWalked(date, id) {
+    let userActivityAll = this.getActivityDataById(id);
```
Comment (catlynbowles) [inline src/Activity.js:22]: great work with adding this new data, Amanda ! 👩🏼‍🎤

File: src/apiCalls.js [hunk 2]
```diff
@@ -18,3 +18,61 @@ export function fetchAllData() {
...
+    'Content-Type': 'application/json',
+  },
+  body: JSON.stringify(dataToTransmit)
+}).then(res => {return res.json()})
+.catch(error => alert('🤡 OOOPSIES! 🤨 There was an error. Please try again. 🤡 '));
+
+  return response;
+}
+
+export function addUserHydrationData(dataToTransmit) {
```
Comment (catlynbowles) [inline src/apiCalls.js:51]: Wow, these posts are ** chef's kiss ** !! 🧑🏼‍🍳

**Review by catlynbowles — APPROVED** [review]

File: dist/index.html [hunk 3]
```diff
@@ -15,9 +15,9 @@
   <body>
     <h1>FitLit</h1>
     <main>
-      <div class="left-side">
+      <div class="left-side" tabindex='0'>
         <section class="info-card">
-          <h2 class="welcome-text" id="welcomeText">Welcome, name!</h2>
+          <h2 role='presentation'class="welcome-text" id="welcomeText">Welcome, name!</h2>
```
Comment (irmakerdem) [inline dist/index.html:20]: ARIA role!! ♿️

File: dist/index.html [hunk 4]
```diff
@@ -26,36 +26,76 @@ <h2 class="welcome-text" id="welcomeText">Welcome, name!</h2>
           <p class="user-data" id="dailyStepGoal">Daily Step Goal:</p>
           <p class="user-data" id="friends">Friends:</p>
         </section>
+        <div class="bottom-left" tabindex='0'>
+          <section class="input-card">
+            <fieldset>
```
Comment (irmakerdem) [inline dist/index.html:31]: Fancy fieldset element 🌾

File: dist/index.html [hunk 5]
```diff
@@ -26,36 +26,76 @@ <h2 class="welcome-text" id="welcomeText">Welcome, name!</h2>
...
+              <legend>Sleep:</legend>
+              <label class="sleep-date" id="sleepDate">Add Date:</label><br />
+              <input class="sleep-date-input" id="sleepDateInput" type="text" placeholder="dd/mm/yyyy" /><br />
```
Comment (irmakerdem) [inline dist/index.html:34]: The placeholder here makes the app so real to what we see on apps in the real world! So cool!! ⏳

File: dist/index.html [hunk 6]
```diff
@@ -26,36 +26,76 @@ <h2 class="welcome-text" id="welcomeText">Welcome, name!</h2>
...
+              <label class="activity-flights" id="activityFlights">Add Flights of Stairs:</label><br />
+              <input class="activity-goal-input" id="activityFlightsInput" type="number" placeholder="5" /><br /><br />
+              <input class="submit-activity" id="submitActivity" type="submit" value="Submit" />
+            </fieldset>
+          </section>
+        </div>
       </div>
-      <div class="right-and-bottom">
-        <div class="right-side">
+      <div class="right-and-bottom" tabindex='0'>
```
Comment (irmakerdem) [inline dist/index.html:68]: WHOA tabindex! 🗂

File: src/scripts.js [hunk 7]
```diff
@@ -168,8 +187,46 @@ function displayStepsInfo(newUser) {
...
+  const sleepQuality = document.getElementById('sleepQualityInput').value;
+  const formattedDate = new Date(sleepDate).toLocaleDateString();
+  let dataToTransmit = { userID: selectedUser.id, date: formattedDate , hoursSlept: sleepAmount , sleepQuality: sleepQuality };
+  var response = addUserSleepData(dataToTransmit).then((res) => getSleepData().then(sleepDataFromApi => console.log(sleepDataFromApi)));;
+}
+
+function addUserHydrationDataFromUserInput() {
+  const hydrationDate = document.getElementById('hydrationDateInput').value;
+  const numberOfOunces = document.getElementById('hydrationOuncesInput').value;
+  const formattedDate = new Date(hydrationDate).toLocaleDateString();
```
Comment (irmakerdem) [inline src/scripts.js:219]: Fancy date method 🎩

**Review by irmakerdem — APPROVED** [review]
Awesome job adding in the input fields, bringing in ARIA roles, and of course getting the POST to work!! 🥳

**Pull request merged**